src/main/java/org/openstreetmap/josm/plugins/pt_assistant/actions/PTMendRelationSupportClasses.java

```diff
@@ -0,0 +1,447 @@
+                        letterY, 25);
+                else
+                    drawFixVariantLetter("W : " + tr(I18N_SOLUTIONS_BASED_ON_OTHER_RELATIONS), Color.PINK, letterX,
+                        letterY, 25);
+                letterY = letterY + 60;
+            }
+
+            for (int i = 0; i < 5; i++) {
+                if (wayColoring.containsValue(chr)) {
+                    drawFixVariantLetter(chr.toString(), colors[i], letterX, letterY, 35);
+                    letterY = letterY + 60;
+                }
+                chr++;
+            }
+
+            /* display the "Esc", "Skip" label: */
+            drawFixVariantLetter("Esc : " + tr(I18N_CLOSE_OPTIONS), Color.WHITE, letterX, letterY, 25);
+
+            letterY = letterY + 60;
+            if (numeric) {
+                drawFixVariantLetter("7 : " + tr(I18N_SKIP), Color.WHITE, letterX, letterY, 25);
+                letterY = letterY + 60;
+                drawFixVariantLetter("8 : " + tr(I18N_BACKTRACK_WHITE_EDGE), Color.WHITE, letterX, letterY, 25);
+                letterY = letterY + 60;
+                drawFixVariantLetter("9 : " + tr(I18N_REMOVE_CURRENT_EDGE), Color.WHITE, letterX, letterY, 25);
+            } else {
+                drawFixVariantLetter("S : " + tr(I18N_SKIP), Color.WHITE, letterX, letterY, 25);
+                letterY = letterY + 60;
+                drawFixVariantLetter("V : " + tr(I18N_BACKTRACK_WHITE_EDGE), Color.WHITE, letterX, letterY, 25);
+                letterY = letterY + 60;
+                drawFixVariantLetter("Q : " + tr(I18N_REMOVE_CURRENT_EDGE), Color.WHITE, letterX, letterY, 25);
+            }
+        }
+
+        /**
+         *
+         */
+        public void drawOptionsToRemoveWays(String notice) {
+            drawFixVariantsWithParallelLines(false);
+            boolean numeric = PTAssistantPluginPreferences.NUMERICAL_OPTIONS.get();
+
+            double letterX = MainApplication.getMap().mapView.getBounds().getMinX() + 20;
+            double letterY = MainApplication.getMap().mapView.getBounds().getMinY() + 100;
+
+            if (notice != null) {
+                drawFixVariantLetter("Error:  " + notice, Color.WHITE, letterX, letterY, 25);
+                letterY = letterY + 60;
+            }
+            if (numeric) {
+                drawFixVariantLetter("1 : " + tr(I18N_REMOVE_WAYS), FIVE_COLOR_PALETTE[0], letterX, letterY, 25);
+                letterY = letterY + 60;
+                drawFixVariantLetter("2 : " + tr(I18N_NOT_REMOVE_WAYS), FIVE_COLOR_PALETTE[1], letterX, letterY, 25);
+                letterY = letterY + 60;
+                drawFixVariantLetter("3 : " + tr(I18N_REMOVE_WAYS_WITH_PREVIOUS_WAY), FIVE_COLOR_PALETTE[4], letterX,
+                    letterY, 25);
+                letterY = letterY + 60;
+                if (notice.equals("vehicle travels against oneway restriction")) {
```

Comment:
This `notice` variable will definitely need reworking. Currently it is used both as a user facing message and as seen here as some kind of state.

This kind of making decisions depending on what message is shown to the user is really bad and at the latest when i18n comes into play things like this fall apart.

The `notice` variable seems to be used for these kinds of messages:
* `vehicle travels against oneway restriction`
* `$str restriction violated` (where `$str` is one of `only_right_turn`, `only_left_turn`, `only_u_turn`, `only_straight_on`, `only_entry`, `only_exit`)
* `Multiple common nodes found between current and next way`